protocol-units/da/movement/celestia/util/src/ir_blob.rs

```diff
@@ -212,13 +212,9 @@ pub mod celestia {
 
 		// todo: it would be nice to have this be self describing over the compression and serialization format
 		fn try_from(blob: CelestiaBlob) -> Result<Self, Self::Error> {
-			// decompress blob.data with zstd
-			let decompressed =
-				zstd::decode_all(blob.data.as_slice()).context("failed to decompress blob")?;
-
-			// deserialize the decompressed data with bcs
-			let blob =
-				bcs::from_bytes(decompressed.as_slice()).context("failed to deserialize blob")?;
+			// decompress the blob and deserialize the data with bcs
+			let decoder = zstd::Decoder::new(blob.data.as_slice())?;
+			let blob = bcs::from_reader(decoder).context("failed to deserialize blob")?;
```

Comment:
Do you know how [`fill_buf`](https://github.com/gyscos/zstd-rs/blob/bfe1e34f593c2427e9dc35f1b98d06788174711f/src/stream/zio/reader.rs#L123) would be called here? I wondering if this in fact going to raise an `Error` or whether it might just panic if the buffer size is restricted.

Reply:
I don't think I fully understand this question, and we didn't restrict the size of the internal buffer.
But you did bring to my attention that the slice also supports `BufRead` and we can take advantage of it.

Reply:
Yes, bcs would return an `Error`.

I have implemented a customizable limit on length of any variable-length fields in https://github.com/zefchain/bcs/pull/12.